src/connector/src/source/kinesis/split.rs

```diff
@@ -18,20 +18,30 @@ use serde::{Deserialize, Serialize};
 use crate::error::ConnectorResult;
 use crate::source::{SplitId, SplitMetaData};
 
+/// See <https://docs.aws.amazon.com/kinesis/latest/APIReference/API_StartingPosition.html> for more details.
 #[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize, Hash)]
 pub enum KinesisOffset {
+    /// Corresponds to `TRIM_HORIZON`. Points the oldest record in the shard.
     Earliest,
+    /// Corresponds to `LATEST`. Points to the (still-nonexisting) record just after the most recent one in the shard.
     Latest,
-    SequenceNumber(String),
+    /// Corresponds to `AFTER_SEQUENCE_NUMBER`. Points the record just after the one with the given sequence number.
+    #[serde(alias = "SequenceNumber")] // for backward compatibility
+    AfterSequenceNumber(String),
+    /// Corresponds to `AT_TIMESTAMP`. Points to the (first) record right at or after the given timestamp.
     Timestamp(i64),
+
     None,
 }
```

Comment:
@tabVersion Let's revisit this enum, let's view it as a correspondence of the starting position types described here https://docs.aws.amazon.com/kinesis/latest/APIReference/API_StartingPosition.html.

Then the enum should be:

```rust
pub enum KinesisOffset {
    TrimHorizon, // TRIM_HORIZON
    Latest, // LATEST
    AtSequenceNumber(String), // AT_SEQUENCE_NUMBER
    AfterSequenceNumber(String), // AFTER_SEQUENCE_NUMBER
    AtTimestamp(i64), // AT_TIMESTAMP

    None,
}
```

Now, the `AtSequenceNumber` variant is actually not used by any of our code, so we can remove it. So it becomes:

```rust
pub enum KinesisOffset {
    TrimHorizon, // TRIM_HORIZON
    Latest, // LATEST
    AfterSequenceNumber(String), // AFTER_SEQUENCE_NUMBER
    AtTimestamp(i64), // AT_TIMESTAMP

    None,
}
```

And we happened to use `Earliest` for `TrimHorizon` and `Timestamp` for `AtTimestamp` for the sake of consistency with other source types. Now it becomes:

```rust
pub enum KinesisOffset {
    Earliest, // TRIM_HORIZON
    Latest, // LATEST
    AfterSequenceNumber(String), // AFTER_SEQUENCE_NUMBER
    Timestamp(i64), // AT_TIMESTAMP

    None,
}
```

Is that clear?